### ISSUE TITLE
Could not pass event PlayerCommandPreprocessEvent to TradePlus

### ISSUE THREAD

**Issue by falceso** [Trophonix/TradePlus#34]
**Plugin Version:** TradePlus v3.2
**How to reproduce:** /trade <player>
**Server Version:** PaperMC 1.13.2
**Issue:** Unable to use Trade GUI. Ex. Clicking on the accept button. The below log appears after both people accept the trade request.

```[01:03:53 ERROR]: Could not pass event PlayerCommandPreprocessEvent to TradePlus v3.2
java.lang.NullPointerException: null
at java.lang.String.replace(String.java:2240) ~[?:1.8.0_202]
at com.trophonix.tradeplus.util.ItemFactory.replaceInMeta(ItemFactory.java:83) ~[?:?]
at com.trophonix.tradeplus.extras.EconomyExtra.getIcon(EconomyExtra.java:38) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.updateExtras(Trade.java:415) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.<init>(Trade.java:81) ~[?:?]
at com.trophonix.tradeplus.commands.TradeCommand.onCommand(TradeCommand.java:120) ~[?:?]
at com.trophonix.tradeplus.commands.CommandHandler.lambda$onCommand$1(CommandHandler.java:36) ~[?:?]
at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_202]
at com.trophonix.tradeplus.commands.CommandHandler.onCommand(CommandHandler.java:35) ~[?:?]
at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor352.execute(Unknown Source) ~[?:?]
at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:67) ~[patched_1.13.2.jar:git-Paper-563]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.13.2.jar:git-Paper-563]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.13.2.jar:git-Paper-563]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:514) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1790) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1606) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:15) ~[patched_1.13.2.jar:git-Paper-563]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_202]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_202]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SystemUtils.java:108) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1011) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:938) ~[patched_1.13.2.jar:git-Paper-563]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:836) ~[patched_1.13.2.jar:git-Paper-563]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]```

**Comment by falceso** [comment]
Seems this issue is still valid.

```> [23:07:34 INFO]: [TradePlus] Initializing economy extra...
> [23:07:34 INFO]: [TradePlus] Initializing experience extra...
> [23:07:34 ERROR]: Could not pass event PlayerCommandPreprocessEvent to TradePlus v3.21
java.lang.NullPointerException: null
at java.lang.String.replace(String.java:2240) ~[?:1.8.0_202]
at com.trophonix.tradeplus.util.ItemFactory.replaceInMeta(ItemFactory.java:85) ~[?:?]
at com.trophonix.tradeplus.extras.EconomyExtra.getIcon(EconomyExtra.java:38) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.updateExtras(Trade.java:415) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.<init>(Trade.java:81) ~[?:?]
at com.trophonix.tradeplus.commands.TradeCommand.onCommand(TradeCommand.java:122) ~[?:?]
at com.trophonix.tradeplus.commands.CommandHandler.lambda$onCommand$1(CommandHandler.java:36) ~[?:?]
at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_202]
at com.trophonix.tradeplus.commands.CommandHandler.onCommand(CommandHandler.java:35) ~[?:?]
at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor416.execute(Unknown Source) ~[?:?]
at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.13.2.jar:git-Paper-588]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:520) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1774) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1590) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:15) ~[patched_1.13.2.jar:git-Paper-588]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_202]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_202]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SystemUtils.java:108) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1011) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:938) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:836) ~[patched_1.13.2.jar:git-Paper-588]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]```

**Comment by Trophonix** [comment]
Oh, maybe your config is missing something? I will check for other possible npe causes in that code tomorrow..

**Comment by falceso** [comment]
I've attempted to re-generate the config files, then the follow error happen on startup.

```> [04:19:40 INFO]: [TradePlus] Enabling TradePlus v3.21
> [04:19:40 INFO]: You appear to be running version 113
> [04:19:40 INFO]: [TradePlus] Corrected gui.acceptid (STAINED_GLASS_PANE:14 -> RED_STAINED_GLASS_PANE)
> [04:19:40 INFO]: [TradePlus] Corrected gui.cancelid (STAINED_GLASS_PANE:13 -> GREEN_STAINED_GLASS_PANE)
> [04:19:40 INFO]: [TradePlus] Corrected gui.separatorid (STAINED_GLASS_PANE:15 -> BLACK_STAINED_GLASS_PANE)
> [04:19:40 ERROR]: Error occurred while enabling TradePlus v3.21 (Is it up to date?)
java.lang.IllegalArgumentException: Path cannot be null
at org.apache.commons.lang.Validate.notNull(Validate.java:192) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.configuration.MemorySection.getDefault(MemorySection.java:763) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.configuration.MemorySection.get(MemorySection.java:217) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.configuration.MemorySection.contains(MemorySection.java:115) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.configuration.MemorySection.contains(MemorySection.java:111) ~[patched_1.13.2.jar:git-Paper-588]
at com.trophonix.tradeplus.TradePlus.onEnable(TradePlus.java:548) ~[?:?]
at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:265) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:334) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:412) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugin(CraftServer.java:443) ~[patched_1.13.2.jar:git-Paper-588]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.enablePlugins(CraftServer.java:357) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.l(MinecraftServer.java:608) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:570) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:431) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.DedicatedServer.init(DedicatedServer.java:316) ~[patched_1.13.2.jar:git-Paper-588]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:786) ~[patched_1.13.2.jar:git-Paper-588]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
> [04:19:40 INFO]: [TradePlus] Disabling TradePlus v3.21```

**Comment by falceso** [comment]
Hello, just wanted to bump this issue. Currently running on the latest version, 3.23 and now items being traded don't even show up in the GUI for the other player.

```
> [16:13:17 ERROR]: Could not pass event PlayerCommandPreprocessEvent to TradePlus v3.23
java.lang.NullPointerException: null
at java.lang.String.replace(String.java:2240) ~[?:1.8.0_202]
at com.trophonix.tradeplus.util.ItemFactory.replaceInMeta(ItemFactory.java:85) ~[?:?]
at com.trophonix.tradeplus.extras.EconomyExtra.getIcon(EconomyExtra.java:38) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.updateExtras(Trade.java:419) ~[?:?]
at com.trophonix.tradeplus.trade.Trade.<init>(Trade.java:81) ~[?:?]
at com.trophonix.tradeplus.commands.TradeCommand.onCommand(TradeCommand.java:122) ~[?:?]
at com.trophonix.tradeplus.commands.CommandHandler.lambda$onCommand$1(CommandHandler.java:37) ~[?:?]
at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_202]
at com.trophonix.tradeplus.commands.CommandHandler.onCommand(CommandHandler.java:36) ~[?:?]
at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor456.execute(Unknown Source) ~[?:?]
at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.13.2.jar:git-Paper-621]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.13.2.jar:git-Paper-621]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66) ~[patched_1.13.2.jar:git-Paper-621]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:520) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1774) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1590) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:15) ~[patched_1.13.2.jar:git-Paper-621]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_202]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_202]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SystemUtils.java:108) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1018) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:940) ~[patched_1.13.2.jar:git-Paper-621]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:837) ~[patched_1.13.2.jar:git-Paper-621]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
```

**Comment by Trophonix** [comment]
Should now be fixed, will release version 3.26 soon.
There must be something wrong with your configs, though. This problem does not exist in trade+ as a whole, and the crashes are happening in places related to config reading. I would try deleting the whole TradePlus folder in plugins and let Trade+ regenerate them completely.
